src/Document/MediaManager.php

```diff
@@ -13,7 +13,7 @@
 
 namespace Sonata\MediaBundle\Document;
 
-use Sonata\DatagridBundle\Pager\PagerInterface;
+use Sonata\AdminBundle\Datagrid\PagerInterface;
```

Comment:
This looks wrong. You should use the bundle without the admin stuff

Reply:
There is no right or wrong.

The DatagridBundle is just a bunch of duplicated code from the SonataAdminBundle. All this code is now outdated as explained here: https://github.com/sonata-project/SonataDatagridBundle/pull/340#issuecomment-864523118. There is multiple misconception in the DatagridBundle: https://github.com/sonata-project/SonataDatagridBundle/issues/235.

Taking a look at all the package using the DatagridBundle, half of them are also using the AdminBundle which means users and with two different PagerInterface, two different ProxyQueryInterface, and so on. The other half of these bundles are still highly related to the AdminBundle and use it as de dev dependency. I cannot be sure it works without the AdminBundle and if it does, I'm cannot sure a lot of developper are using it without the AdminBundle.

Making the DatagridBundle up-to-date would require a lot of changes, which means a lot of contributor time and a new major version. Looking at the activity of the contributor and the maintainer, we don't have this time. Releasing the AdminBundle 4 was hard enough.

It's better to stop the mess with the DatagridBundle, so we're planning to deprecate it to be more realistic with the time we have. Less package will provide a better quality. Requiring the AdminBundle instead of the DatagridBundle is not a big change. It will even ends with smaller vendors for package witch were requiring both.

Reply:
I can understand, that there's something wrong and that there are some misconceptions between some bundle, but using admin related classes inside entity repositories looks more wrong than before.

I'm a huge fan of [deptrac](https://github.com/qossmic/deptrac) to define layers that should not communitate to each other. Maybe be should define rules which community is allowed by design and which isn't. The entity -> admin dependency is a no go.